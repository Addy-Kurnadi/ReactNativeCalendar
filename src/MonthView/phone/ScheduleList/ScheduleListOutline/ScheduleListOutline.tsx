import * as React from "react"
import { Text, View } from "react-native"
import styles from "./SchedulteListOutline.styles"

interface Props {
    height: number
    numberToRender: number
}

export const individualHourlyBox = (height: number = 0, numberToRender: number = 0) => {
    const calcHeight = height / 5
    const halfCalcHeight = height / 10
    const rows = []
    for (let i = 0; i < numberToRender; i++) {
        rows.push(
            <View key={i} style={[styles.borderLeft, styles.borderStyle, {height}]}>
                <View style={[
                    styles.borderCircle,
                    {
                        height: calcHeight,
                        width: calcHeight,
                        borderRadius: halfCalcHeight,
                        top: -halfCalcHeight,
                        left: -halfCalcHeight
                    }
                ]}/>
            </View>
        )
    }
    return rows
}

export const ScheduleListOutline = ({height, numberToRender}: Props) => {
    const halfOfHeight = height / 2
    const calcHeight = height / 5
    const halfCalcHeight = height / 10
    return (
        <View style={styles.outerViewContainer}>
            <View style={[styles.borderLeft, styles.borderStyle, {height: halfOfHeight}]}/>
                {individualHourlyBox(height, numberToRender)}
            <View style={[styles.borderLeft, {height: halfOfHeight}]}>
                <View style={[
                    styles.borderCircle,
                    {
                        height: calcHeight,
                        width: calcHeight,
                        borderRadius: halfCalcHeight,
                        top: -halfCalcHeight,
                        left: -halfCalcHeight
                    }
                ]}/>
            </View>
        </View>
    )
}
