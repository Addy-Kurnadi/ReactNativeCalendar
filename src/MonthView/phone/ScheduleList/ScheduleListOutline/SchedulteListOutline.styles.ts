import { StyleSheet, TextStyle, ViewStyle } from "react-native"

interface Style {
    outerViewContainer: ViewStyle
    headerViewContainer: ViewStyle
    textViewContainer: ViewStyle
    emptyViewContainer: ViewStyle
    borderStyle: ViewStyle
    borderLeft: ViewStyle
    borderCircle: ViewStyle
    textStyle: TextStyle
}

export default StyleSheet.create<Style>({
    headerViewContainer: {
        flex: 1,
        height: 50
    },
    emptyViewContainer: {
        flex: 1
    },
    textViewContainer: {
        flex: 4
    },
    outerViewContainer: {
        flex: 4
    },
    borderStyle: {
        borderStyle: "solid",
        borderBottomWidth: 1,
        borderBottomColor: "#d3d3d3"
    },
    borderLeft: {
        borderStyle: "solid",
        borderLeftWidth: 1,
        borderLeftColor: "#d3d3d3"
    },
    borderCircle: {
        backgroundColor: "white",
        borderWidth: 1,
        borderColor: "#d3d3d3",
        borderStyle: "solid",
        position: "absolute"
    },
    textStyle: {
        fontSize: 16,
        fontWeight: "bold",
        color: "black"
    }
})
