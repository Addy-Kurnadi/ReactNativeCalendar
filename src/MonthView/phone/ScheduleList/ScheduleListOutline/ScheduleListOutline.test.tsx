import * as React from "react"
import { create } from "react-test-renderer"
import { individualHourlyBox, ScheduleListOutline } from "./ScheduleListOutline"

describe("<Timeline />", () => {
    it("should render with given props", () => {
        const component = create(
            <ScheduleListOutline
                height={50}
                numberToRender={10}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
    it("should trigger individualHourlyBox function to render jsx", () => {
        const component = individualHourlyBox(50, 10)
        expect(component.length).toBe(10)

        const component2 = individualHourlyBox()
        expect(component2.length).toBe(0)
    })
})
