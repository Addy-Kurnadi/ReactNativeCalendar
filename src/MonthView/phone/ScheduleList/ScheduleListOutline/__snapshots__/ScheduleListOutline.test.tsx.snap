// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render with given props 1`] = `
<View
  style={
    Object {
      "flex": 4,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 25,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "borderBottomColor": "#d3d3d3",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#d3d3d3",
          "borderLeftWidth": 1,
          "borderStyle": "solid",
        },
        Object {
          "height": 25,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#d3d3d3",
            "borderStyle": "solid",
            "borderWidth": 1,
            "position": "absolute",
          },
          Object {
            "borderRadius": 5,
            "height": 10,
            "left": -5,
            "top": -5,
            "width": 10,
          },
        ]
      }
    />
  </View>
</View>
`;
